internal/provider/resource_bgp_session.go

```diff
@@ -394,6 +437,9 @@ func extractBgpSession(d *schema.ResourceData) packetfabric.BgpSession {
 	if md5, ok := d.GetOk("md5"); ok {
 		bgpSession.Md5 = md5.(string)
 	}
+	for _, nat := range d.Get("nat").(*schema.Set).List() {
```

Comment:
Does this code work properly when `nat` is not present? I would expect it to fail as `Get` will return `nil`. It is safer to use `GetOk` (so the loop will be guarded by `if`).

Reply:
1. added check `if nat, ok := d.GetOk("nat"); ok {`
2. also added `,omitempty` in struct definition for `Nat` 

However, it looks like I am still sending the `nat{}` :(

```
{"md5": "", "l3_address": "169.254.51.2/29", "address_family": "v4", "remote_address": "169.254.51.1/29", "remote_asn": 64534, "multihop_ttl": 1, "orlonger": true, "prefixes": [{"prefix": "0.0.0.0/0", "type": "in"}, {"prefix": "0.0.0.0/0", "type": "out"}], "nat": {}}
```

Reply:
I got it working using a pointer as suggested.